akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -153,45 +154,48 @@ private[remote] class Association(
 
   @volatile private[this] var _outboundControlIngress: OptionVal[OutboundControlIngress] = OptionVal.None
   @volatile private[this] var materializing = new CountDownLatch(1)
-  @volatile private[this] var changeOutboundCompression: Vector[ChangeOutboundCompression] = Vector.empty
+  @volatile private[this] var outboundCompressionAccess: Vector[OutboundCompressionAccess] = Vector.empty
   // in case there is a restart at the same time as a compression table update
   private val changeCompressionTimeout = 5.seconds
 
   private[remote] def changeActorRefCompression(table: CompressionTable[ActorRef]): Future[Done] = {
     import transport.system.dispatcher
-    val c = changeOutboundCompression
+    val c = outboundCompressionAccess
     val result =
       if (c.isEmpty) Future.successful(Done)
       else if (c.size == 1) c.head.changeActorRefCompression(table)
       else Future.sequence(c.map(_.changeActorRefCompression(table))).map(_ ⇒ Done)
-    timeoutAfter(result, changeCompressionTimeout, new ChangeOutboundCompressionFailed)
+    timeoutAfter(result, changeCompressionTimeout, new AccessOutboundCompressionFailed)
   }
   // keyed by stream queue index
   private[this] val streamMatValues = new AtomicReference(Map.empty[Int, OutboundStreamMatValues])
   private[this] val idle = new AtomicReference[Option[Cancellable]](None)
 
   private[remote] def changeClassManifestCompression(table: CompressionTable[String]): Future[Done] = {
     import transport.system.dispatcher
-    val c = changeOutboundCompression
+    val c = outboundCompressionAccess
     val result =
       if (c.isEmpty) Future.successful(Done)
       else if (c.size == 1) c.head.changeClassManifestCompression(table)
       else Future.sequence(c.map(_.changeClassManifestCompression(table))).map(_ ⇒ Done)
-    timeoutAfter(result, changeCompressionTimeout, new ChangeOutboundCompressionFailed)
+    timeoutAfter(result, changeCompressionTimeout, new AccessOutboundCompressionFailed)
   }
 
   private def clearOutboundCompression(): Future[Done] = {
     import transport.system.dispatcher
-    val c = changeOutboundCompression
+    val c = outboundCompressionAccess
     val result =
       if (c.isEmpty) Future.successful(Done)
       else if (c.size == 1) c.head.clearCompression()
       else Future.sequence(c.map(_.clearCompression())).map(_ ⇒ Done)
-    timeoutAfter(result, changeCompressionTimeout, new ChangeOutboundCompressionFailed)
+    timeoutAfter(result, changeCompressionTimeout, new AccessOutboundCompressionFailed)
   }
 
   private def clearInboundCompression(originUid: Long): Unit =
-    transport.inboundCompressions.foreach(_.close(originUid))
+    transport.inboundCompressionAccess match {
```

Comment:
I made it an OptionVal some time ago, so foreach is not an option. You're right it's not hot-patch. Should I revert it to Option?